appinventor/components/src/com/google/appinventor/components/runtime/Serial.java

```diff
@@ -0,0 +1,109 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2016-2019 MIT, All rights reserved
+// Copyright 2017-2019 Kodular, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.runtime;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.physicaloid.lib.Physicaloid;
+
+import com.google.appinventor.components.common.ComponentCategory;
+import com.google.appinventor.components.annotations.DesignerComponent;
+import com.google.appinventor.components.annotations.SimpleEvent;
+import com.google.appinventor.components.annotations.SimpleFunction;
+import com.google.appinventor.components.annotations.SimpleObject;
+import com.google.appinventor.components.annotations.UsesLibraries;
+import com.google.appinventor.components.common.YaVersion;
+
+import java.io.UnsupportedEncodingException;
+
+@DesignerComponent(version = YaVersion.SERIAL_COMPONENT_VERSION,
+    description = "Serial component which can be used to connect to devices like Arduino",
+    category = ComponentCategory.CONNECTIVITY,
+    nonVisible = true,
+    iconName = "images/arduino.png")
+
+@SimpleObject
+@UsesLibraries(libraries = "physicaloid.jar")
+public class Serial extends AndroidNonvisibleComponent implements Component {
+  private static final String LOG_TAG = "Serial Component";
+
+  private Context context;
+
+  Physicaloid mPhysicaloid;
+
+  private int baudRate = 9600;
+
+  public Serial(ComponentContainer container) {
+    super(container.$form());
+    context = container.$context();
+    Log.d(LOG_TAG, "Created");
+  }
+
+  @SimpleFunction(description = "Initializes serial connection.")
+  public void InitializeSerial() {
+    mPhysicaloid = new Physicaloid(context);
```

Comment:
Might make sense to call `setBaudrate` here just in case the default changes in a future version.

Reply:
Done.